app/models/mixins/retirement_mixin.rb

```diff
@@ -10,9 +10,9 @@ module RetirementMixin
   end
 
   module ClassMethods
-    def make_retire_request
-      options = {:task => task, :userid => current_user, :ids => objs, :src_ids => objs}
-      (klass.to_s + "RetireRequest").constantize.make_request(@request_id, options, current_user)
+    def make_retire_request(*options)
+      options = {:src_ids => options.present? ? options.first : id, :src_klass => options.present? ? self : name}
```

Comment:
> `options.present? ? self : name`

Why `self` vs `name`?  Shouldn't it always be `name`?

Reply:
it's bimodal, so the UI retirement is always name but we call make_retire_request without options from the mixin in the other case which has the model name as self.